### PR TITLE
Add intel from explore

### LINKED ISSUE #1856: Add IOCs to a sketch from the Explore view
**Is your feature request related to a problem? Please describe.**
It would be nice to be able to "save" IOCs that we encounter during an investigation to a local, sketch-specific, threat intelligence repository.

**Describe the solution you'd like**
* Regular expressions that scan events for common types of IOCs
* Select text and save it as an IOC

### PR DESCRIPTION
**Pull request by tomchop**
+ What existing problem does this PR solve?

Currently unable to curate a list of interesting IOCs or artifacts in a sketch.

+ What new feature is being introduced with this PR?

With this PR, a user can click on highlighted IOCs in the explore view and "save" them as local threat intelligence IOCs. These are associated to a sketch.

+ Overview of changes to existing functions if required.

* New context menu Vue components
* New dependency on https://github.com/AlbertLucianto/vue-text-highlight, which highlights and gives extra functionality to text in components.
* Upon click on a highlighted IOC in the event attributes, a tooltip is displayed for IOC curation (see demo video)

**Checks**

- [ ] All tests succeed.
- [ ] Unit tests added.
- [ ] e2e tests added.
- [ ] Documentation updated.

**Closing issues**

Closes #1856

**Demo**

intelligence

### CODE REVIEW

File: timesketch/frontend/src/components/Common/TsIOCMenu.vue [hunk 1]
```diff
@@ -0,0 +1,184 @@
+<!--
+Copyright 2019 Google Inc. All rights reserved.
```
Comment (berggren) [inline timesketch/frontend/src/components/Common/TsIOCMenu.vue:2]:
```suggestion
Copyright 2021 Google Inc. All rights reserved.
```

File: timesketch/frontend/src/components/Common/TsIOCMenu.vue [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+export default {
+  components: {
+    TsContextMenu,
+  },
+  props: ['attributeKey'],
+  name: 'TsIOCMenu',
+  data() {
+    return {
+      IOCTypes: [
+        { regex: /[0-9]{1,3}(\.[0-9]{1,3}\.)/g, type: 'ip' },
```
Comment (berggren) [inline timesketch/frontend/src/components/Common/TsIOCMenu.vue:68]: No IPv6? :)

File: timesketch/frontend/src/components/Common/TsIOCMenu.vue [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            position: 'is-top',
+            actionText: 'View intelligence',
+            indefinite: false,
+            onAction: () => {
+              this.$router.push({ name: 'Intelligence' })
+            },
+          })
+          // refresh computed property based on attributes
+          this.$store.state.meta.attributes = attributes
+          console.log('Attribute added successfully')
```
Comment (berggren) [inline timesketch/frontend/src/components/Common/TsIOCMenu.vue:139]: Drop the console.log

File: timesketch/frontend/src/components/Explore/EventListRowDetail.vue [hunk 4]
```diff
@@ -88,6 +105,17 @@ export default {
       }
       this.$emit('addChip', chip)
     },
+    handleSelectionChange(event) {
+      if (event.target.closest('.ioc-match') || event.target.closest('.ioc-context-menu')) {
+        return
+      }
+      console.log('selection change')
```
Comment (berggren) [inline timesketch/frontend/src/components/Explore/EventListRowDetail.vue:112]:
```suggestion

```

File: timesketch/frontend/src/components/Explore/EventListRowDetail.vue [hunk 5]
```diff
@@ -88,6 +105,17 @@ export default {
...
+      const text = window.getSelection().toString()
+      this.regexes.selection = text
+      if (this.regexes.selection !== '') {
+        console.log(text)
```
Comment (berggren) [inline timesketch/frontend/src/components/Explore/EventListRowDetail.vue:116]: Is this a left over from development?

File: timesketch/frontend/src/views/Intelligence.vue [hunk 6]
```diff
@@ -22,34 +22,104 @@ limitations under the License.
...
+        label: 'Local intelligence',
+        deletable: true,
+      },
+    }
+  },
+  methods: {
+    localIntelligenceDeleteCallback(ioc) {
+      const data = this.localIntelligence.data.filter(i => i.ioc !== ioc.ioc)
+      ApiClient.addSketchAttribute(this.sketch.id, 'intelligence_local', { data: data }, 'intelligence').then(() => {
+        console.log(`${ioc.ioc} deleted successfully`)
```
Comment (berggren) [inline timesketch/frontend/src/views/Intelligence.vue:96]: Same here, I guess this is left over from development?

File: timesketch/frontend/src/components/Common/TsIOCMenu.vue [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+      return this.$store.state.sketch
+    },
+  },
+  mounted() {},
+}
+</script>
+
+<style lang="scss">
+.ioc-match {
+  background-color: #ddd;
```
Comment (berggren) [inline timesketch/frontend/src/components/Common/TsIOCMenu.vue:155]: Does this look good in dark mode as well? :)

File: timesketch/frontend/src/components/Common/TsIOCMenu.vue [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+        }
+        attributes.intelligence_local.value.data.push(ioc)
+        ApiClient.addSketchAttribute(
+          this.sketch.id,
+          'intelligence_local',
+          attributes.intelligence_local.value,
+          'intelligence'
+        ).then(() => {
+          Snackbar.open({
+            message: 'Attribtue added succesfully',
```
Comment (berggren) [inline timesketch/frontend/src/components/Common/TsIOCMenu.vue:128]:
```suggestion
            message: 'Attribute added successfully',
```

File: timesketch/frontend/src/components/Common/TsIOCMenu.vue [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+            type: 'is-white',
```
Comment (berggren) [inline timesketch/frontend/src/components/Common/TsIOCMenu.vue:129]: Let's ceck dark mode here as well. But we can also leave it, seeing that I'm doing an overhaul of the dark mode in an upcoming PR

**Review by berggren — CHANGES_REQUESTED** [review]
First round - @tomchop PTAL

File: timesketch/frontend/src/components/Common/TsIOCMenu.vue (hunk 2 above)
Comment (tomchop) [reply timesketch/frontend/src/components/Common/TsIOCMenu.vue:68]: Not yet :) I feel the reflex is a tad more complicated

File: timesketch/frontend/src/components/Common/TsIOCMenu.vue (hunk 7 above)
Comment (tomchop) [reply timesketch/frontend/src/components/Common/TsIOCMenu.vue:155]: It doesn't look great, changed.

**Comment by tomchop** [comment]
@berggren back to you now!

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**